internal/cmd/kubernetes.go

```diff
@@ -193,26 +193,42 @@ func writeKubeconfig(user *api.User, destinations []api.Destination, grants []ap
 		}
 	}
 
-	configPath := defaultConfig.ConfigAccess().GetDefaultFilename()
+	configFile := defaultConfig.ConfigAccess().GetDefaultFilename()
 
-	configDir := filepath.Dir(configPath)
+	return safelyWriteConfigToFile(kubeConfig, configFile)
+}
+
+// safelyWriteConfigToFile creates a temp file, then overwrites the target
+func safelyWriteConfigToFile(kubeConfig clientcmdapi.Config, fileToWrite string) error {
+	// get the directory of the file we're writing to avoid cross-filesystem moves
+	configDir := filepath.Dir(fileToWrite)
 	if err := os.MkdirAll(configDir, os.ModePerm); err != nil {
 		return err
 	}
 
+	temp, err := os.Create(fmt.Sprintf("%s/infra-kube-config-%d", configDir, time.Now().Unix()))
```

Comment:
Doesn't `ioutil.TempFile` return the same type (`os.File`) ?  Is there something about the temp file that doesn't work?